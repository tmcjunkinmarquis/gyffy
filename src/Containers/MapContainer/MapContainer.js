import React, { Component } from 'react';
import { apiKey } from '../../apiKey.json';
import { Map,
  Marker,
  AdvancedMarkerElement,
  GoogleApiWrapper } from 'google-maps-react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
// import { setCenter } from '../../Actions/Actions';

export class MapContainer extends Component {
  constructor(props){
    super(props);
    this.state = {
      position:{}
    };
  }

  selectedStationMarker = ()=>{
    return <Marker
      name={this.props.selectedStation.name}
      position={{
        lat: this.props.selectedStation.latitude, 
        lng: this.props.selectedStation.longitude 
      }} />;
  }
  
  componentDidMount() {
    this.setState({position: {
      lat: 39.7512822,
      lng: -104.9944365
    }});
    const lat = this.props.selectedStation.latitude;
    const lng = this.props.selectedStation.longitude;
  }
  
  render() {
    const style = {
      'margin-top': '-80px',
      width: '100vw',
      height: '65vh',
      position: 'relative'
    };
    const detailsStyle = {
      margin: '30px'
    };
    const initialCenter = {
      lat: 38.902775,
      lng: -77.455649
    };
    const center = {
      lat: this.props.selectedStation.latitude,
      lng: this.props.selectedStation.longitude
    };

    return (
      < Map
        google={this.props.google}
        style={style}
        initialCenter={initialCenter}
        center={center}
        zoom={15}
      >
        {this.selectedStationMarker()}
        <div style={detailsStyle}>
          <h3>{this.props.selectedStation.name}</h3>
          <h4>{this.props.selectedStation.phone}</h4>
        </div>
      </Map>  
    );
  }
}

export const mapStateToProps = (state) => ({
  selectedStation: state.selectedStation
});

export const mapDispatchToProps = (dispatch) =>({
  /* setCenter: (center) => dispatch(setCenter(center)) */
});

console.log("apiKey:", apiKey)
const googleWrapper = GoogleApiWrapper({
  apiKey: apiKey["apiKey"]
})(MapContainer);

export default connect(mapStateToProps, mapDispatchToProps)(googleWrapper);

MapContainer.propTypes = {
  google: PropTypes.object,
  loaded: PropTypes.bool,
  selectedStation: PropTypes.object,
  setCenter: PropTypes.object
};  
