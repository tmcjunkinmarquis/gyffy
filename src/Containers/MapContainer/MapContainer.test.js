import React from 'react';
import { shallow } from 'enzyme';
import { MapContainer, mapStateToProps } from '../MapContainer/MapContainer';

describe('MapContainer Container', () => {

  describe('MapContainer Component', () => {
    let wrapper;
    let mockProps;
    beforeEach(()=>{
      mockProps = {
        selectedStation: { name: 'bubbas', latitude: 38.802775, longitude: -103.9944365}
      }
      wrapper = shallow(
      <MapContainer
      {...mockProps} />)
    })

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    });
  });

  describe('mapStateToProps', () => {
    let wrapper;
    let mockProps;
    beforeEach(() => {
      mockProps = {
        selectedStation: { name: 'bubbas', latitude: 38.802775, longitude: -103.9944365 }
      }
      wrapper = shallow(
        <MapContainer
          {...mockProps} />)
    })

    it('should return an object with keys selectedStation', () => {
      const mockState = {
        selectedStation: { name: 'bubbas', latitude: 38.802775, longitude: -103.9944365 }
      };
      const expected = {
        selectedStation: { name: 'bubbas', latitude: 38.802775, longitude: -103.9944365 }
      };
      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    });
  }); 

  
});