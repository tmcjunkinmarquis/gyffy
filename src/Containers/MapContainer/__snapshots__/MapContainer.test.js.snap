// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapContainer Container MapContainer Component should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MapContainer
    selectedStation={
        Object {
            "latitude": 38.802775,
            "longitude": -103.9944365,
            "name": "bubbas",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "center": Object {
        "lat": 38.802775,
        "lng": -103.9944365,
      },
      "centerAroundCurrentLocation": false,
      "children": <Marker
        name="bubbas"
        position={
                Object {
                        "lat": 38.802775,
                        "lng": -103.9944365,
                      }
        }
/>,
      "containerStyle": Object {},
      "google": undefined,
      "initialCenter": Object {
        "lat": 38.902775,
        "lng": -77.455649,
      },
      "style": Object {
        "height": "50vh",
        "margin": "auto",
        "position": "relative",
        "width": "90vw",
      },
      "visible": true,
      "zoom": 15,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "name": "bubbas",
        "position": Object {
          "lat": 38.802775,
          "lng": -103.9944365,
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "center": Object {
          "lat": 38.802775,
          "lng": -103.9944365,
        },
        "centerAroundCurrentLocation": false,
        "children": <Marker
          name="bubbas"
          position={
                    Object {
                              "lat": 38.802775,
                              "lng": -103.9944365,
                            }
          }
/>,
        "containerStyle": Object {},
        "google": undefined,
        "initialCenter": Object {
          "lat": 38.902775,
          "lng": -77.455649,
        },
        "style": Object {
          "height": "50vh",
          "margin": "auto",
          "position": "relative",
          "width": "90vw",
        },
        "visible": true,
        "zoom": 15,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "name": "bubbas",
          "position": Object {
            "lat": 38.802775,
            "lng": -103.9944365,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
